### PR TITLE
IPv4 host network addresses convert with htonl() not htons()

### PR DESCRIPTION
**Pull request by nbriggs**
In the host based TCP subrs, converting a FIXP to an IPv4 address was never going to work calling htons(), since addresses are 32-bit not 16-bit.  Use htonl().  This looks like a day-0 bug.

### CODE REVIEW

**Comment by nbriggs** [comment]
Don't merge this until the previous PR, #500  has been merged

**Review by waywardmonkeys — APPROVED** [review]

**Comment by waywardmonkeys** [comment]
Curious why this depends on #500.

**Comment by nbriggs** [comment]
@waywardmonkeys - you're right, they are independent.  I forgot that I *had* switched back to master before creating the 2nd one.

**Pull request merged**
